hack/generate/samples/ansible/memcached.go

```diff
@@ -0,0 +1,254 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ansible
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	kbtestutils "sigs.k8s.io/kubebuilder/test/e2e/utils"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/operator-framework/operator-sdk/hack/generate/samples/pkg"
+	"github.com/operator-framework/operator-sdk/test/utils"
+	testutils "github.com/operator-framework/operator-sdk/test/utils"
+)
+
+// MemcachedAnsible defines the context for the sample
+type MemcachedAnsible struct {
+	ctx *pkg.SampleContext
+}
+
+// NewMemcachedAnsible return a MemcachedAnsible
+func NewMemcachedAnsible(ctx *pkg.SampleContext) MemcachedAnsible {
+	return MemcachedAnsible{ctx}
+}
+
+// Prepare prepare the directory for the Memcached Ansible Sample
+func (ma *MemcachedAnsible) Prepare() {
+	log.Infof("destroying directory for memcached Ansible samples")
+	ma.ctx.Destroy()
+
+	log.Infof("creating directory for Ansible Sample")
+	err := ma.ctx.Prepare()
+	pkg.CheckError("error to creating directory for Ansible Sample", err)
+
+	log.Infof("setting domain and GKV")
+	ma.ctx.Domain = "example.com"
+	ma.ctx.Version = "v1alpha1"
+	ma.ctx.Group = "cache"
+	ma.ctx.Kind = "Memcached"
+}
+
+// Run the steps to create the Memcached Ansible Sample
+func (ma *MemcachedAnsible) Run() {
```

Comment:
The Run has all steps to gen the samples. The sample can be checked in the /testdata/ansible
@asmacdo, @jmrodri , @theishshah, @varshaprasad96